packages/libs/src/sdk/services/Solana/programs/ClaimableTokensClient/ClaimableTokensClient.ts

```diff
@@ -195,4 +226,43 @@ export class ClaimableTokensClient extends BaseSolanaProgramClient {
       claimableTokensPDA: this.authorities[mint]
     })
   }
+
+  /**
+   * Override the sendTransaction method to provide some more friendly errors
+   * back to the consumer for ClaimableTokens instructions
+   */
+  public override async sendTransaction(
+    transaction: Transaction | VersionedTransaction,
+    sendOptions?: SendTransactionOptions | undefined
+  ): Promise<string> {
+    try {
+      return await super.sendTransaction(transaction, sendOptions)
+    } catch (e) {
+      if (e instanceof SendTransactionError) {
+        try {
+          const error = CustomInstructionError.parseSendTransactionError(e)
+          if (error) {
+            const instructions = await this.getInstructions(transaction)
+            const instruction = instructions[error.instructionIndex]
+            if (instruction && instruction.programId.equals(this.programId)) {
+              const decodedInstruction =
+                ClaimableTokensProgram.decodeInstruction(instruction)
+              throw new ClaimableTokensError({
+                code: error.code,
+                instructionName:
+                  ClaimableTokensInstruction[
+                    decodedInstruction.data.instruction
+                  ] ?? 'Unknown',
+                cause: e
+              })
+            }
+          }
+        } catch (e) {
+          // If failed to provide user friendly error, surface original error
+          this.logger.warn('Failed to parse ClaimableTokensError error', e)
```

Comment:
why not error? because it throws below?

Reply:
Yes. This should not mask the true error - bubble up the original error